src/platform/libretro/libretro.c

```diff
@@ -516,58 +648,53 @@ void retro_unload_game(void) {
 }
 
 size_t retro_serialize_size(void) {
-	return core->stateSize(core);
+	return 4 * (core1->stateSize(core1));
 }
 
 bool retro_serialize(void* data, size_t size) {
 	if (size != retro_serialize_size()) {
 		return false;
 	}
-	core->saveState(core, data);
+	void *tmp;
+	size_t tmp_size = size / 4;
+
+	tmp = malloc(tmp_size);
+	core1->saveState(core1, tmp);
+	memcpy(data, tmp, tmp_size);
+	free(tmp);
+	tmp = malloc(tmp_size);
```

Comment:
can't say freeing and immediately allocating a new one makes much sense, but I guess it does no harm either?

Reply:
ah yeah considering it's the same size you're right